### ISSUE TITLE
Why feature normalization required on MNIST 

### ISSUE THREAD

**Issue by LasaniHussain** [rasbt/python-machine-learning-book-2nd-edition#68]
Sir, I have one doubt. The MNIST dataset has 784 columns corresponding to the unrolled 784 pixel values.Each pixel(column) has the same range (0-255). I remember you mentioned in the chapter 4 Building good data sets : that feature normalization helps in faster convergence when all features are in the same scale.In MNIST they are already on the same scale but you still normalized them in Chapter 13 where you used keras for MINST prediction. I tried without normalizing pixel values to [-1,1] range and surprisingly the gradient descent didn't converge even after 50 epochs on the same net architecture.

**Comment by rasbt** [comment]
> In MNIST they are already on the same scale but you still normalized them in Chapter 13 where you used keras for MINST prediction. I tried without normalizing pixel values to [-1,1] range and surprisingly the gradient descent didn't converge even after 50 epochs on the same net architecture.

Having relatively small weights also helps with numerical stability. However, you can also get relatively ok results with the [0, 255] pixel range using the same architecture. However, for this, you need to change the learning rate accordingly.
